internal/engine/errors/errors.go

```diff
@@ -204,11 +204,11 @@ func ErrorAsAlertStatus(err error) db.AlertStatusTypes {
 
 // AlertStatusAsError returns the error for a given alert status
 func AlertStatusAsError(prevStatus *db.ListRuleEvaluationsByProfileIdRow) error {
-	if prevStatus == nil || !prevStatus.AlertStatus.Valid {
+	if prevStatus == nil {
 		return errors.New("no previous alert state")
 	}
 
-	s := prevStatus.AlertStatus.AlertStatusTypes
```

Comment:
Since we no longer use left joins, these values are no longer nullable. The changes in #4197 handles the case where there is no prior state.